File: python/store.py
import pickle

file_name = "pickle.pkl"


def save(todo_list):
    with open(file_name, 'wb') as f:
        pickle.dump(todo_list, f)


def fetch():
    with open(file_name, 'rb') as f:
        todo_list = pickle.load(f)
        return todo_list

File: ruby/store.rb
require 'json'

$store_file_filename = 'todo.json'

def read_from_file
    begin
        storeFile = File.new($store_file_filename, 'r')
        return JSON.load storeFile
    rescue
        return []
    end
end

def persistent_to_file todo_list
    File.open($store_file_filename, 'w') do |file|
      return JSON.dump(todo_list, file)
    end
end

def prune
    File.delete($store_file_filename) if File.exist?($store_file_filename)
end
